pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -513,16 +513,41 @@ export default class SettingsValidator {
     return errors.length === 0 && changed;
   }
 
-  protected checkStringArray<S>(mergedSettings: S, currentValue: string[], desiredValue: string[], errors: string[], fqname: string): boolean {
+  protected checkUniqueStringArray<S>(mergedSettings: S, currentValue: string[], desiredValue: string[], errors: string[], fqname: string): boolean {
     if (!Array.isArray(desiredValue) || desiredValue.some(s => typeof (s) !== 'string')) {
       errors.push(this.invalidSettingMessage(fqname, desiredValue));
 
       return false;
     }
+    const duplicateValues = this.findDuplicates(desiredValue);
+
+    if (duplicateValues.length > 0) {
+      duplicateValues.sort((a: string, b:string) => {
+        return a.localeCompare(b);
+      });
+      errors.push(`field '${ fqname }' has duplicate entries: "${ duplicateValues.join('", "') }"`);
+
+      return false;
+    }
 
     return currentValue.length !== desiredValue.length || currentValue.some((v, i) => v !== desiredValue[i]);
   }
 
+  protected findDuplicates(list: string[]): string[] {
+    if (list.length <= 1) {
+      return [];
+    }
+    const first = list[0];
+
+    list = list.slice(1);
+    if (list.includes(first)) {
+      // Filter out `first` so we don't count it again as a duplicate.
+      return [first].concat(this.findDuplicates(list.filter(elt => elt !== first)));
+    } else {
+      return this.findDuplicates(list);
+    }
+  }
```

Comment:
If we use `Set` we can drop the recursion:
```suggestion
  /** Find any entries that are duplicated, in arbitrary order. */
  protected findDuplicates(list: string[]): string[] {
    const seen = new Set<string>();
    const dupes = new Set<string>();
    
    for (const elem of list) {
      if (!seen.has(elem)) {
        seen.add(elem);
      } else {
        dupes.add(elem);
      }
    }

    return Array.from(dupes);
  }
```
If we don't want to keep two sets:
```suggestion
  /** Find any entries that are duplicated, in arbitrary order. */
  protected findDuplicates(list: string[]): string[] {
    const counts: Record<string, number> = {};
    
    for (const elem of list) {
      counts[elem] = (counts[elem] ?? 0) + 1;
    }
    
    return Object.entries(counts)
      .filter(([, v]) => v > 1)
      .map(([k]) => k);
  }
```

Reply:
The set code is simpler